src/caretogether-pwa/src/Referrals/PartneringFamilyScreen.tsx

```diff
@@ -80,6 +82,68 @@ export function PartneringFamilyScreen() {
     ? `${partneringFamily.family?.adults!.filter(adult => adult.item1!.id === partneringFamily.family!.primaryFamilyContactPersonId)[0]?.item1?.lastName} Family`
     : "...");
 
+  //#region Arrangement Filtering
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const handleFilterClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleFilterClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'Filter' : undefined;
+
+  enum ArrangementFilterType {
+    Active = "Active",
+    Cancelled = "Cancelled"
+  }
+  const defaultArrangementTypes = [ArrangementFilterType.Active];
+  const [arrangementFilterOptions, setArrangementFilterOptions] = useState<IFilterOption[]>(
+    Object.keys(ArrangementFilterType).map(k => {
+      return {
+        key: k,
+        text: k,
+        selected: defaultArrangementTypes.includes(k as ArrangementFilterType)
+      }
+    })
+  );
+
+  const filterArrangements = (event: ChangeEvent<HTMLInputElement>, checked: boolean) => {
+    setArrangementFilterOptions(arrangementFilterOptions.map(o => {
+      if (o.text === event.target.name) {
+        o.selected = !o.selected;
+      }
+      return o;
+    }));
+  }
+
+  const meetsArrangementFilterCriteria = (arrangement: Arrangement): boolean => {
+    // TODO: Reduce the number of times this method fires on the initial page load
+    let result = false;
+    const selectedOptions = arrangementFilterOptions.filter(o => o.selected).map(o => o.text);    
+    if (selectedOptions.length) {
```

Comment:
I think this `if` is unnecessary -- if `.length` is zero, `.forEach(...)` will just exit without doing anything.

Reply:
Fixed